pkg/credential/token.go

```diff
@@ -30,12 +35,49 @@ type GcpTokenFetcher struct {
 	Aud string
 }
 
+// AwsTokenFetcher implements the token fetcher in AWS.
+// It gets the EC2 instance identity document as described
+// in the doc: http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html
+type AwsTokenFetcher struct {
+}
+
+// IsOnAWS returns if node agent is running on AWS EC2 instance
+func (fetcher *AwsTokenFetcher) IsOnAWS() bool {
+	sess := session.Must(session.NewSession())
+	svc := ec2metadata.New(sess)
+	return svc.Available()
+}
+
+// FetchToken fetches the instance identity document as a string
+// Note: currently AWS Golang SDK does not verify the signature,
+// so this function is not very secure. We expect this bug to
+// be fixed in upstream.
+func (fetcher *AwsTokenFetcher) FetchToken() (string, error) {
+	sess := session.Must(session.NewSession())
+	svc := ec2metadata.New(sess)
+	if svc.Available() {
+		doc, err := svc.GetInstanceIdentityDocument()
+		if err == nil {
+			bytes, _ := json.Marshal(doc)
+			return string(bytes), nil
+		}
```

Comment:
do you want to have a else block to handle err != nil case?

Reply:
if `err != nil` it will fall back to end of the function which throws an error.

Reply:
But you will lose the debugging info in 'err', right?

Reply:
good catch! I'll change it